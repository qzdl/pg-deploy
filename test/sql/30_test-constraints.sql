--                                      ██                    ██            ██
--                                     ░██                   ░░            ░██
--   █████   ██████  ███████   ██████ ██████ ██████  ██████   ██ ███████  ██████  ██████
--  ██░░░██ ██░░░░██░░██░░░██ ██░░░░ ░░░██░ ░░██░░█ ░░░░░░██ ░██░░██░░░██░░░██░  ██░░░░
-- ░██  ░░ ░██   ░██ ░██  ░██░░█████   ░██   ░██ ░   ███████ ░██ ░██  ░██  ░██  ░░█████
-- ░██   ██░██   ░██ ░██  ░██ ░░░░░██  ░██   ░██    ██░░░░██ ░██ ░██  ░██  ░██   ░░░░░██
-- ░░█████ ░░██████  ███  ░██ ██████   ░░██ ░███   ░░████████░██ ███  ░██  ░░██  ██████
--  ░░░░░   ░░░░░░  ░░░   ░░ ░░░░░░     ░░  ░░░     ░░░░░░░░ ░░ ░░░   ░░    ░░  ░░░░░░

-- CONSTRAINTS

BEGIN;
SET client_min_messages TO WARNING;
CREATE EXTENSION pgdeploy;

CREATE SCHEMA testp;
CREATE SCHEMA testr;
CREATE TABLE testp.con(
    i INT CONSTRAINT yeah CHECK (i>0),
    ii INT CHECK (ii > i),
    iii INT CHECK (0>iii));

CREATE TABLE testr.con(
    i INT CONSTRAINT hmm CHECK(i>ii),
    ii INT,
    iii INT CHECK (0>iii));

SELECT conname, pg_get_constraintdef(c.oid) AS constrainddef
FROM pg_constraint c
WHERE conrelid IN (
    SELECT attrelid FROM pg_attribute
    WHERE attrelid IN (
        SELECT c.oid
        FROM pg_class c
        INNER JOIN pg_namespace n on n.oid = c.relnamespace
        WHERE c.relname = 'con'
            AND n.nspname IN ('testp', 'testr'))
        AND attname='tableoid');

SELECT * FROM pgdeploy.reconcile_constraints(
  'testp'::NAME, 'con'::NAME,'testp.con'::regclass::oid
    ,'testr'::NAME, 'con'::NAME,'testp.con'::regclass::oid)
ORDER BY reconcile_constraints DESC;


-- CLEAN UP
DROP EXTENSION pgdeploy CASCADE;
ROLLBACK;
