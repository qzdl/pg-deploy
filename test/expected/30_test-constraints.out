--                                      ██                    ██            ██
--                                     ░██                   ░░            ░██
--   █████   ██████  ███████   ██████ ██████ ██████  ██████   ██ ███████  ██████  ██████
--  ██░░░██ ██░░░░██░░██░░░██ ██░░░░ ░░░██░ ░░██░░█ ░░░░░░██ ░██░░██░░░██░░░██░  ██░░░░
-- ░██  ░░ ░██   ░██ ░██  ░██░░█████   ░██   ░██ ░   ███████ ░██ ░██  ░██  ░██  ░░█████
-- ░██   ██░██   ░██ ░██  ░██ ░░░░░██  ░██   ░██    ██░░░░██ ░██ ░██  ░██  ░██   ░░░░░██
-- ░░█████ ░░██████  ███  ░██ ██████   ░░██ ░███   ░░████████░██ ███  ░██  ░░██  ██████
--  ░░░░░   ░░░░░░  ░░░   ░░ ░░░░░░     ░░  ░░░     ░░░░░░░░ ░░ ░░░   ░░    ░░  ░░░░░░
-- CONSTRAINTS
BEGIN;
SET client_min_messages TO WARNING;
CREATE EXTENSION pgdeploy;
ERROR:  syntax error at or near "CREATE"
CREATE SCHEMA testp;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
CREATE SCHEMA testr;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
CREATE TABLE testp.con(
    i INT CONSTRAINT yeah CHECK (i>0),
    ii INT CHECK (ii > i),
    iii INT CHECK (0>iii));
ERROR:  current transaction is aborted, commands ignored until end of transaction block
CREATE TABLE testr.con(
    i INT CONSTRAINT hmm CHECK(i>ii),
    ii INT,
    iii INT CHECK (0>iii));
ERROR:  current transaction is aborted, commands ignored until end of transaction block
SELECT conname, pg_get_constraintdef(c.oid) AS constrainddef
FROM pg_constraint c
WHERE conrelid IN (
    SELECT attrelid FROM pg_attribute
    WHERE attrelid IN (
        SELECT c.oid
        FROM pg_class c
        INNER JOIN pg_namespace n on n.oid = c.relnamespace
        WHERE c.relname = 'con'
            AND n.nspname IN ('testp', 'testr'))
        AND attname='tableoid');
ERROR:  current transaction is aborted, commands ignored until end of transaction block
SELECT * FROM pgdeploy.reconcile_constraints(
  'testp'::NAME, 'con'::NAME,'testp.con'::regclass::oid
    ,'testr'::NAME, 'con'::NAME,'testp.con'::regclass::oid)
ORDER BY reconcile_constraints DESC;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
-- CLEAN UP
DROP EXTENSION pgdeploy CASCADE;
ERROR:  current transaction is aborted, commands ignored until end of transaction block
ROLLBACK;
